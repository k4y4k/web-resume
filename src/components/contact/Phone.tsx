import * as React from 'react'
import { graphql, useStaticQuery } from 'gatsby'
import { FiPhone } from '@react-icons/all-files/fi/FiPhone'

interface PhoneTypes {
  restrictDisplay?: boolean
}

const Phone = ({ restrictDisplay = true }: PhoneTypes): JSX.Element | null => {
  if (restrictDisplay) return null

  const { phone } = useStaticQuery(graphql`
    {
      num: file(extension: { eq: "json" }, name: { eq: "data" }) {
        childDataJson {
          basics {
            phone
          }
        }
      }
    }
  `).num.childDataJson.basics

  // xxx xxx xxxx
  const formattedNum = Array.from(phone)
  formattedNum.splice(3, 0, ' ')
  formattedNum.splice(7, 0, ' ')

  return (
    <li data-testid='contactPhone'>
      <FiPhone /> {formattedNum.join('')}
    </li>
  )
}

export default Phone
