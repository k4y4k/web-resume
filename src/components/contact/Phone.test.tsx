import * as React from 'react'
import { render, screen } from '@testing-library/react'
import Phone from './Phone'

describe('<Phone />', () => {
  test('handles no phone data', () => {
    render(<Phone num='' />)

    const phone = screen.queryByTestId('contactPhone')
    expect(phone).not.toBeInTheDocument()
    expect(phone).toMatchSnapshot()
  })

  test('hides phone number by default', () => {
    render(<Phone num='0123456789' />)

    const phone = screen.queryByTestId('contactPhone')
    expect(phone).not.toBeInTheDocument()
    expect(phone).toMatchSnapshot()
  })

  test('displays phone number', () => {
    render(<Phone num='0123456789' restrictDisplay={false} />)

    const phone = screen.getByTestId('contactPhone')
    expect(phone).toHaveTextContent('012 345 6789')
    expect(phone).toMatchSnapshot()
  })
})
