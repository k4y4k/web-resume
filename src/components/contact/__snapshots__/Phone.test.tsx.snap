// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Phone /> displays phone number 1`] = `
<p
  data-testid="contactPhone"
>
  012 345 6789
</p>
`;

exports[`<Phone /> handles no phone data 1`] = `null`;

exports[`<Phone /> hides phone number by default 1`] = `null`;
